vmdb/lib/vmdb/configuration_encoder.rb

```diff
@@ -19,6 +19,23 @@ def self.symbolize!(h)
       h.each_key { |k| h[k].symbolize_keys! }.symbolize_keys!
     end
 
+    class YamlLoader < Psych::Visitors::ToRuby
+      NUMBER_WITH_METHOD_REGEX = /^([0-9\.,]+)\.([a-z]+)$/
+
+      def visit_Psych_Nodes_Scalar(node)
+        if node.value =~ NUMBER_WITH_METHOD_REGEX
+          if $2 == 'percent'
+            super
```

Comment:
You're right!  I totally read the original implementation incorrectly.  I'll fix this, but I'm not too keen on this PR right now, so I'll change the title to WIP.